### PR TITLE
Add re2c package, change php-8.3 to not build from release.

### PR DESCRIPTION
**Pull request by vaikas**
In order to build from the source, need the re2c package, so add that too. I'll follow up with changing php-8.2 later with this style.

Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

**Review by rawlingsj — DISMISSED** [review]

**Comment by vaikas** [comment]
Hm, failed twice with the same error, so maybe not a flake? 🤔

**Comment by rawlingsj** [comment]
Yeah - odd that it works for the arm build thouhj

**Comment by rawlingsj** [comment]
tl;dr to get past the failure in the presubmit we can add wget as an explicit build dependency to override wget from busybox.

---

Managed to build this locally after explicitly adding wget as a build dependency.  

Had a bit of a dig around and found that wget is also provided by busybox
```
apk info --who-owns /usr/bin/wget

/usr/bin/wget symlink target is owned by busybox-1.36.1-r2
```

I'm not sure why busybox's wget seems to fail on x86 and not arm yet.  However to reproduce the error, we can run two containers, one arm, the second amd
```
docker run --privileged --platform linux/aarch64 --rm -it -v "${PWD}:/work" -w /work cgr.dev/chainguard/wolfi-base:latest
```
```
docker run --privileged --platform linux/amd64 --rm -it -v "${PWD}:/work" -w /work cgr.dev/chainguard/wolfi-base:latest
```

In each of them run `wget pear.php.net`

The results are different

arm:
```
wget pear.php.net
Connecting to pear.php.net (109.203.101.62:80)
Connecting to pear.php.net (109.203.101.62:443)
wget: note: TLS certificate validation not implemented
saving to 'index.html'
```

amd:
```
Connecting to pear.php.net (109.203.101.62:80)
Connecting to pear.php.net (109.203.101.62:443)
wget: note: TLS certificate validation not implemented
wget: TLS error from peer (alert code 40): handshake failure
wget: error getting response: Connection reset by peer
```

**Comment by rawlingsj** [comment]
https://github.com/wolfi-dev/os/issues/9187

**Comment by kaniini** [comment]
The fetch pipeline pulls wget in automatically.  If we are not using fetch anymore, but the build system depends on wget's presence, then we should add it as a dependency.

**Review by imjasonh — APPROVED** [review]
